addons/product/models/product_template.py

```diff
@@ -693,14 +691,12 @@ def _create_variant_ids(self):
                 variants_to_create += current_variants_to_create
                 variants_to_activate += current_variants_to_activate
 
-            else:
-                for variant in existing_variants.values():
-                    is_combination_possible = self._is_combination_possible_by_config(
-                        combination=variant.product_template_attribute_value_ids,
-                        ignore_no_variant=True,
-                    )
-                    if is_combination_possible:
-                        current_variants_to_activate += variant
+            elif existing_variants:
+                # THIS is wrong and broken
+                variants_combinations = Product.concat(*existing_variants.values()).product_template_attribute_value_ids
+                current_variants_to_activate += Product.concat(*[existing_variants[possible_combination]
+                    for possible_combination in tmpl_id._filter_combinations_impossible_by_config(variants_combinations, ignore_no_variant=True)
+                ])
```

Comment:
This breaks the existing behavior (see new test).
`variants_combinations`  will hold all the ptavs of all the existing variants, in one recordset, you're losing the consideration of the different combinations here.